CleanSwift/DetailModule/DetailViewController.swift

```diff
@@ -0,0 +1,151 @@
+// DetailViewController.swift
+// Copyright © SolovevAA. All rights reserved.
+
+import StoreKit
+import UIKit
+
+/// Основной экран
+final class DetailViewController: UIViewController {
+    // MARK: - Private Constants
+
+    private enum Constants {
+        static let idAppStoreNumber = 586_447_913
+    }
+
+    // MARK: - Private Visual Properties
+
+    private lazy var changeTextButton: UIButton = {
+        let button = UIButton()
+        button.setTitle(Text.changeText, for: .normal)
+        button.backgroundColor = .lightGray
+        button.layer.cornerRadius = 15
+        button.addTarget(self, action: #selector(changeTextButtonAction(sender:)), for: .touchUpInside)
+        return button
+    }()
+
+    private lazy var deepLinkButton: UIButton = {
+        let button = UIButton()
+        button.setTitle(Text.deepLinkText, for: .normal)
+        button.backgroundColor = .lightGray
+        button.layer.cornerRadius = 15
+        button.addTarget(self, action: #selector(deepLinkButtonAction(sender:)), for: .touchUpInside)
+        return button
+    }()
+
+    private let titleNasaLabel: UILabel = {
+        let label = UILabel()
+        label.textAlignment = .center
+        label.font = UIFont.boldSystemFont(ofSize: 22)
+        return label
+    }()
+
+    // MARK: - Private Properties
+
+    private var interactor: DetailBusinessLogic
+    private var router: DetailRoutingLogic
+
+    // MARK: - Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        initForm()
+    }
+
+    // MARK: - Initializers
+
+    init(interactor: DetailBusinessLogic, router: DetailRoutingLogic) {
+        self.interactor = interactor
+        self.router = router
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError(Text.initErrorText)
+    }
+
+    // MARK: - Private Methods
+
+    private func initForm() {
+        interactor.requestInitForm(DetailModel.InitForm.Request())
+    }
+
+    @objc private func changeTextButtonAction(sender: UIButton) {
+        interactor.requestTextChanged(DetailModel.TextChanged.Request())
+    }
+
+    @objc private func deepLinkButtonAction(sender: UIButton) {
+        guard let url = URL(string: Text.deepLinkUrlText) else { return }
+        if UIApplication.shared.canOpenURL(url) {
+            UIApplication.shared.open(url)
+        } else {
+            let storeProductViewController = SKStoreProductViewController()
+            storeProductViewController
+                .loadProduct(
+                    withParameters: [
+                        SKStoreProductParameterITunesItemIdentifier:
+                            NSNumber(value: Constants.idAppStoreNumber)
+                    ]
+                )
+            present(storeProductViewController, animated: true)
```

Comment:
```suggestion
    @objc private func changeTextButtonAction(sender: UIButton) {
        interactor.requestTextChanged(DetailModel.TextChanged.Request())
    }

    @objc private func deepLinkButtonAction(sender: UIButton) {
        guard let url = URL(string: Text.deepLinkUrlText) else { return }
        if UIApplication.shared.canOpenURL(url) {
            UIApplication.shared.open(url)
        } else {
            let storeProductViewController = SKStoreProductViewController()
            storeProductViewController
                .loadProduct(
                    withParameters: [
                        SKStoreProductParameterITunesItemIdentifier:
                            NSNumber(value: Constants.idAppStoreNumber)
                    ]
                )
            present(storeProductViewController, animated: true)
```
@objc либо в самый верх подними либо вниз класса опусти